### PR TITLE
Merge Rampup into Main

### PR DESCRIPTION
**Pull request by shimkiv**

### CODE REVIEW

**Comment by mitschabaude** [comment]
@shimkiv thanks for getting this started! make sure to also do the same merge in `o1js-bindings` and include here

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,7 +17,9 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased](https://github.com/o1-labs/snarkyjs/compare/b1d8d5910...HEAD)
 
-> No unreleased changes yet
+### Breaking changes
+
+- Changes to verification keys caused by updates to the proof system. This breaks all deployed contracts https://github.com/o1-labs/snarkyjs/pull/1016
```
Comment (mitschabaude) [inline CHANGELOG.md:22]:
```suggestion
- Changes to verification keys caused by updates to the proof system. This breaks all deployed contracts https://github.com/o1-labs/o1js/pull/1016 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,9 +15,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
     _Security_ in case of vulnerabilities.
  -->
 
-## [Unreleased](https://github.com/o1-labs/o1js/compare/fbd4b2717...HEAD)
+## [0.12.3](https://github.com/o1-labs/o1js/compare/fbd4b2717...HEAD)
```
Comment (mitschabaude) [inline CHANGELOG.md:18]: could we make this follow the standard and include a specific commit instead of HEAD?

Comment (shimkiv) [reply CHANGELOG.md]: Sure, thank you.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "o1js",
   "description": "TypeScript framework for zk-SNARKs and zkApps",
-  "version": "0.12.2",
+  "version": "0.12.3",
```
Comment (Trivo25) [inline package.json:4]: should this be a major bump since its not backwards compatible with already compiled contract/programs (verification key changes)?

**Review by Trivo25 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (shimkiv) [reply package.json]: Done.

**Pull request merged**